packages/runtime-core/src/hydration.ts

```diff
@@ -130,15 +136,20 @@ export function createHydrationFunctions(
           }
         } else {
           if ((node as Text).data !== vnode.children) {
-            hasMismatch = true
-            __DEV__ &&
-              warn(
-                `Hydration text mismatch:` +
-                  `\n- Client: ${JSON.stringify((node as Text).data)}` +
-                  `\n- Server: ${JSON.stringify(vnode.children)}`
-              )
-            ;(node as Text).data = vnode.children as string
+            if (
+              decodeHtml((node as Text).data) !==
+              decodeHtml(vnode.children as string)
+            ) {
+              hasMismatch = true
+              __DEV__ &&
+                warn(
+                  `Hydration text mismatch:` +
+                    `\n- Client: ${JSON.stringify((node as Text).data)}` +
```

Comment:
In the hydration phase, the `node` is actually returned by the Server, and the `vnode` is generated by the client.
I feel this should be:
```
`\n- Server: ${JSON.stringify((node as Text).data)}` +
`\n- Client: ${JSON.stringify(vnode.children)}`
```

Reply:
👌